src/nativestore/core.cljs

```diff
@@ -233,93 +238,73 @@
   IPrintWithWriter
   (-pr-writer [native writer opts]
     (-write writer "#native ")
-    (print-map
-     (->> (js-keys native)
-          (keep (fn [k]
-                  (let [v (aget native k)
-                        k (keyword k)]
-                    (when-not (or (fn? v)
-                                  (#{:cljs$lang$protocol_mask$partition0$ :cljs$lang$protocol_mask$partition1$ :__ro :nativestore$core$IReadOnly$} k))
-                      [k v])))))
-     pr-writer writer opts))
+    (print-map (map (fn [k] [k (aget native (name k))]) @keyset)
+               pr-writer writer opts))
 
   ICounted
   (-count [native]
-    (alength (js-keys native)))
+    (count @keyset))
 
   ILookup
   (-lookup [native k]
     (-lookup native k nil))
   (-lookup [native k not-found]
-    (let [key (if (keyword? k) (name k) k)
-          res (aget native key)]
-      (cond (nil? res)
-            not-found
-            (array? res)
-            (if (satisfies? IReference (aget res 0))
-              (amap res i ret (resolve-ref (aget res i)))
-              res)
-            (satisfies? IReference res)
-            (resolve-ref res)
-            :default
-            res)))
+    (let [v (aget native (name k))]
+      (cond
+        (nil? v) not-found
+        (array? v) (if (satisfies? IReference (aget v 0))
+                     (amap v i ret (resolve-ref (aget v i)))
+                     v)
+        (satisfies? IReference v) (resolve-ref v)
+        :default v)))
 
   ITransientAssociative
   (-assoc! [native k v]
     (when (and (-read-only? native) (not *transaction*))
       (throw (js/Error. "Cannot mutate store values outside transact!: ")))
-    (aset native (if (keyword? k) (name k) k) v)
+    (vswap! keyset conj k)
+    (aset native (name k) v)
```

Comment:
This and all other places this change is made breaks the existing contract that (seq (assoc! (native) :foo 10)) => [[:foo 10]].  This changes results in (seq (assoc! (native) :foo 10)) => [["foo" 10]].  Native objects purposefully stored keywords as keywords so we didn't end up with ambiguities with maps of strings to value vs. maps of keywords to values.